### DISCUSSION TITLE
Hide realm from endpoints, JWT issuer, URLs advertised in the well-known endpoint etc

### DISCUSSION THREAD

**Discussion by nishanthreddy** [keycloak/keycloak#12154] [Q&A]
Is there a way to hide realm from endpoints, JWT issuer, URLs advertised in the well-known endpoint etc?

Instead of
https://mydomain.com/realms/master/protocol/openid-connect/token -> https://mydomain.com/protocol/openid-connect/token

Likewise the response of .well-known/openid-configuration should not have realm
{
issuer: "https://mydomain.com/<del>realms/master</del>,
authorization_endpoint: "https://mydomain.com/<del>realms/master</del>/protocol/openid-connect/auth",
token_endpoint: "https://mydomain.com/<del>realms/master</del>/protocol/openid-connect/token”,
….
}

Issued JWT:
{
"exp": 1652818065,
"iat": 1652817765,
"iss": "https://mydomain.com/<del>realms/master</del>”,
….
}

FrontendURL and the various related settings would not accomplish removing or /realm/{realmname} pattern from the url and it doesn’t seem feasible to achieve this without requiring edits in the Keycloak source code.

**Comment by abstractj** [comment]
Hi @nishanthreddy I don't think that's possible today. Although there are some alternatives to that:

1. Restrict the access to those URLs. That can be achieved with a reverse proxy, or firewall. For example, https://docs.nginx.com/nginx/admin-guide/security-controls/configuring-http-basic-authentication/#combining-basic-authentication-with-access-restriction-by-ip-address.
2. Disable the master realm and define administrator accounts in a realm with less predictable name.

I hope it helps.
